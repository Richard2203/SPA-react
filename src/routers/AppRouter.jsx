import React, { useContext } from 'react';
// Las rutas padres/principales se manejan con BrowserRouter
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import PrivateRoute from '../auth/PrivateRoute';
import PublicRoute from '../auth/PublicRoute';
import { AuthContext } from '../auth/AuthContext';
import LoginScreen from '../components/login/LoginScreen';
import DashboardRouter from './DashboardRouter';

// Por convencion el archivo AppRouter sera el archivo de rutas principal
const AppRouter = () => {
	const { user } = useContext(AuthContext);

	return (
		// el archivo de rutas principal contiene la etiqueta Router
		<Router>
			{/* el contenido entre el div y Switch se volvera contenido o estilos 
			globales que apareceran en todos los componentes*/}
			<div>
				<Switch>
					<PublicRoute
						exact
						path="/login"
						component={LoginScreen}
						isAuthenticated={user.logged}
					/>

					<PrivateRoute
						path="/"
						isAuthenticated={user.logged}
						component={DashboardRouter}
					/>
				</Switch>
			</div>
		</Router>
	);
};

export default AppRouter;
