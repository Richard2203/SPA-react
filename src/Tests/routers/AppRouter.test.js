import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('Pruebas en el componente AppRouter', () => {
	test('debe renderizar el login si no esta autenticado', () => {
		const contextValue = {
			user: { logged: false },
			dispatch: jest.fn(),
		};

		// hacemos uso de un componente de orden superior por lo cual requerimos
		// de mount en lugar de shallow

		// AppRouter hace uso de AuthContext y que esta siendo usada por un
		// useContext por lo cual para enviarle la informacion se pasa mediante
		// AuthContext.Provider
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
	});

	test('debe renderizar todos los componentes si esta autenticado', () => {
		const contextValue = {
			user: { logged: true, name: 'paquito' },
			dispatch: jest.fn(),
		};

		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper.find('nav').exists()).toBe(true);
	});
});
