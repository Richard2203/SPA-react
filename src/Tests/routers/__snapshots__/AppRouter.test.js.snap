// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter debe renderizar el login si no esta autenticado 1`] = `
<div>
  <div>
    <h2>
      LoginScreen
    </h2>
    <hr />
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
